common.sh

```diff
@@ -695,6 +703,59 @@ add_pkg_to_kickstart() {
     return 0
 }
 
+attach_file() {
+    #
+    # Attach a file to module result
+    #
+    # Usage:
+    #
+    #    attach_file SRC [NAME]
+    #
+    # Store file SRC with the module result as NAME.  If NAME is not provided,
+    # last element of the SRC path is used.  SRC cannot be a directory.
+    #
+    # This is somewhat equivalent to `cp SRC /module/result/path/NAME`, except
+    # that the destination path is decided by the API and created as needed.
+    #
+    # Files attached this way may be referenced in solution text directly by
+    # name NAME.  For example:
+    #
+    #     attach_file /etc/resolv.conf
+    #     solution_fmt p "please review [resolv.conf](resolv.conf)"
+    #
+    # would add a hint to solution text including a link 'resolv.conf' to this
+    # file attached by attach_file() command..
+    #
+    local src=$1        # SRC file
+    local dst=$2        # NAME
+    local full_dst      # final destination path (incl. file part)
+    log_debug "attach_file(): src='$src'"
+    log_debug "attach_file(): dst='$dst'"
+    test -n "$src" || {
+        log_error "usage: attach_file SRC [NAME]"
+        exit_error
+    }
+    test -f "$src" || {
+        log_error "no such file: $src"
+        exit_error
+    }
+    case $dst in
+        /*)
+            log_error "destination name must not start with slash: $dst"
+            exit_error
+            ;;
+        "")
+            full_dst="$_MODULE_STORAGE/files/${src##*/}"
+            ;;
+        *)
+            full_dst="$_MODULE_STORAGE/files/$dst"
+            ;;
+    esac
+    log_debug "attach_file(): full_dst='$full_dst'"
+    mkdir -p "${full_dst%/*}"
+    cp "$src" "$full_dst"
```

Comment:
use rather `cp -a` to preserve mode, selinux context, ...

Reply:
Agree. See https://github.com/upgrades-migrations/preupgrade-assistant/pull/271 for details.

Reply:
Already done in next version, though. (GitHub should now be marking this thread as outdated).